### PR TITLE
modified model QuestionAnswer to limit question to 20 characters

### LINKED ISSUE #317: Character-Limit for Question Input field
As a reporter
I want the Question for Chatbot field limited to 20 characters (and renamed Question-Button)
in order to fullfill the facebook-messenger limitation for buttons.

### PR DESCRIPTION
**Pull request by mz99**
My first pull request on this project. Modified the the model QuestionAnswer and the model unit spec. Wasn't sure if the question button still needs to be renamed?

EDIT:
close #317

### CODE REVIEW

File: features/text_components/character_limit_for_question_answers.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  in order to fullfill the facebook-messenger limitation for buttons.
+
+
+  Background:
+    Given that someone is querying to the chatbot
+
+  Scenario:
+    Given I am writing a query to the chatbox
+    I enter a message that is maximum 20 letters long
+    I click on submit
```
Comment (roschaefer) [inline features/text_components/character_limit_for_question_answers.feature]:
```gherkin
Then there should be an expectation for the cucumber scenario
```

Comment (roschaefer) [inline features/text_components/character_limit_for_question_answers.feature]: check why our build server (Travis CI) is failing: https://travis-ci.org/roschaefer/story.board/builds/230438820?utm_source=github_status&utm_medium=notification

Comment (roschaefer) [inline features/text_components/character_limit_for_question_answers.feature]: It's failing at the command `bundle exec cucumber --profile default`.
1. Reproduce the error, ie. run the command on your machine
2. Fix it locally
3. Commit
4. Push
5. Hope for a green build

Comment (roschaefer) [inline features/text_components/character_limit_for_question_answers.feature]: oh, and when you run `bundle exec cucumber` you can copy paste the generated step definitions into `features/steps/step_definitions.rb`

**Comment by drjakob** [comment]
@mz99 Can we resolve this last little thing please?

**Comment by roschaefer** [comment]
@drjakob the list of all `question_answers` with more than 20 characters in the `question` field:

```
Keine Milch
TEST Milchquanti
Stiller Stall
Fühle mich schlapp.
Kuschelige Temperaturen
Kuschelige Temperaturen
Kuschelige Temperaturen
Kälte
CHATBOT TEST noise
CHATBOT TEST noise
Fieber unter normal!
Ziemlich aktiv - bis zuletzt
Hallo Melkstand, da bin ich wieder!
CHATBOT: Erstes Melken Biestmilch
Die Brunst der anderen
Flotzmaul abputzen und weiter.
Meine Milch schmeckt nicht salzig
Keime? Bei uns doch nicht.
Unser Roboter zählt Zellen
Unser Roboter zählt Zellen
Milch geben ist Leistungssport
Was wird eigentlich aus der Milch?
Zeit für Sauberkeit
Das große Schrubben
Zeitvertreib
```

You can see it on the command line `bin/rails c` with 
```
QuestionAnswer.includes(:text_component).all.select{|qa| ! qa.valid?}.each{|qa| puts qa.text_component.heading}
```

**Comment by roschaefer** [comment]
Nope, @drjakob the list of text components is the list of now invalid data in the database. If sb. edits these text component, he/she will get a validation error even if a changed attribute was sth. else

**Comment by drjakob** [comment]
So, what to do now to solve this?

**Comment by roschaefer** [comment]
Nothing. All I could do is to truncate all questions.

**Comment by drjakob** [comment]
Yes, please do so.

**Pull request merged**